ludwig/schema/encoders/image_encoders.py

```diff
@@ -284,368 +284,828 @@ class Stacked2DCNNEncoderConfig(BaseEncoderConfig):
     )
 
-    height: int = schema_utils.NonNegativeInteger(
-        default=None,
-        description="Height of the input image.",
-        parameter_metadata=ENCODER_METADATA["MLPMixerEncoder"]["height"],
+
+@register_encoder_config("inceptionv3_torch", IMAGE)
+@dataclass
+class TVInceptionV3EncoderConfig(TVBaseEncoderConfig):
+    type: str = schema_utils.StringOptions(
+        ["inceptionv3_torch"],
+        default="inceptionv3_torch",
+        allow_none=False,
+        description="Type of encoder.",
     )
 
-    width: int = schema_utils.NonNegativeInteger(
-        default=None,
-        description="Width of the input image.",
-        parameter_metadata=ENCODER_METADATA["MLPMixerEncoder"]["width"],
+    model_variant: Optional[str] = schema_utils.StringOptions(
+        ["base"],
+        default="base",
+        allow_none=False,
+        description="Pretrained model variant to use.",
+        parameter_metadata=ENCODER_METADATA["TVGoogLeNetEncoder"]["model_variant"],
     )
 
-    num_channels: int = schema_utils.NonNegativeInteger(
-        default=None,
-        description="Number of channels to use in the encoder. ",
-        parameter_metadata=ENCODER_METADATA["MLPMixerEncoder"]["num_channels"],
+
+@register_encoder_config("maxvit_torch", IMAGE)
+@dataclass
+class TVMaxVitEncoderConfig(TVBaseEncoderConfig):
+    type: str = schema_utils.StringOptions(
+        ["maxvit_torch"],
+        default="maxvit_torch",
+        allow_none=False,
+        description="Type of encoder.",
     )
 
-    patch_size: int = schema_utils.PositiveInteger(
-        default=16,
-        description="The image patch size. Each patch is patch_size² pixels. Must evenly divide the image width and "
-        "height.",
-        parameter_metadata=ENCODER_METADATA["MLPMixerEncoder"]["patch_size"],
+    model_variant: Optional[str] = schema_utils.StringOptions(
+        ["t"],
+        default="t",
+        allow_none=False,
+        description="Pretrained model variant to use.",
+        parameter_metadata=ENCODER_METADATA["TVMNASNetEncoder"]["model_variant"],
     )
 
-    embed_size: int = schema_utils.PositiveInteger(
-        default=512,
-        description="The patch embedding size, the output size of the mixer if avg_pool is true.",
-        parameter_metadata=ENCODER_METADATA["MLPMixerEncoder"]["embed_size"],
+
+@register_encoder_config("mnasnet_torch", IMAGE)
+@dataclass
+class TVMNASNetEncoderConfig(TVBaseEncoderConfig):
+    type: str = schema_utils.StringOptions(
+        ["mnasnet_torch"],
+        default="mnasnet_torch",
+        allow_none=False,
+        description="Type of encoder.",
     )
 
-    token_size: int = schema_utils.PositiveInteger(
-        default=2048,
-        description="The per-patch embedding size.",
-        parameter_metadata=ENCODER_METADATA["MLPMixerEncoder"]["token_size"],
+    model_variant: Optional[str] = schema_utils.StringOptions(
```

Comment:
Want to double-check if this field is really `Optional`? If so, make sure `allow_none` is `True`.

Reply:
removed optional